go.mod

```diff
@@ -1,10 +1,13 @@
 module github.com/kanisterio/kanister
 
-go 1.19
+go 1.21
```

Comment:
result of `go mod tidy` 

`go` complains when pinning go version to `1.19` 
```shell
╰─❯ go mod tidy -go 1.19
go: github.com/kopia/kopia@v0.13.1-0.20230605143225-3551f743d762 requires go@1.21, but 1.19 is requested
```

Reply:
However, `make build-controller` failed due to
```shell
  ⨯ build failed after 5s                    error=hook failed: go mod download: exit status 1; output: go: errors parsing go.mod:
/go/src/github.com/kanisterio/kanister/go.mod:5: unknown directive: toolchain
```

Reply:
Failed when pinning `go` version without running `go mod tidy` (because that would cause for `go` version to change and the addition of `toolchain go1.21.1` 
```
stale go.mod and go.sum: please run `go mod tidy`
make[1]: *** [Makefile:243: run] Error 1
make[1]: Leaving directory '/home/runner/work/kanister/kanister'
make: *** [Makefile:324: go-mod-tidy] Error 2
Error: Process completed with exit code 2.
```

Reply:
What I will try next is to roll `go` version in `kastenhq` kopia branch https://github.com/kastenhq/kopia/compare/kasten...kastenhq:kopia:kopia-at-edfa9ee-minus-32193f7